core/src/main/java/hudson/model/queue/MappingWorksheet.java

```diff
@@ -121,7 +121,7 @@ private ExecutorChunk(List<ExecutorSlot> base, int index) {
             assert !base.isEmpty();
             computer = base.get(0).getExecutor().getOwner();
             node = computer.getNode();
-            nodeAcl = node.getACL();
+            nodeAcl = node != null ? node.getACL() : null;
```

Comment:
We want to really think here if `nodeACL` is really ok being null or we should throw and illegal state exception

Reply:
The constructor is private, so it is better to propagate the error via checked exceptions.
Use-case: node gets deleted in a race condition. We would rather partially fail the executor mapping so that the queue can reassign tasks on the next iteration